### PR TITLE
Update git-branch-name-generator extension

### PR DESCRIPTION
**Pull request by devjiwonchoi**
> Port of #12780

cc @pernielsentikaer

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [ ] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by devjiwonchoi** [comment]
@michaelbonner Let's confirm the change and update the pics in metadata!

**Comment by mil3na** [comment]
Hi! Is this PR ready for review? @michaelbonner do you want to take a look first?

File: extensions/git-branch-name-generator/src/index.tsx [hunk 1]
```diff
@@ -1,56 +1,73 @@
...
+        <Form.Dropdown.Item value="refactor/" title="refactor (improve efficiency)" />
+        <Form.Dropdown.Item value="chore/" title="chore (non-production changes)" />
+        <Form.Dropdown.Item value="docs/" title="docs (add or modify documentation)" />
       </Form.Dropdown>
-      <Form.TextField title="Branch name" placeholder="Change the photo" {...itemProps.branch_name} />
+      <Form.TextField id="description" title="Description" placeholder="Description of this branch" storeValue />
+      <Form.TextField
+        id="prefix"
+        title="Prefix"
+        placeholder="Add a custom prefix for the branch: (e.g. jiwon)"
```
Comment (michaelbonner) [inline extensions/git-branch-name-generator/src/index.tsx:50]: Can we make the e.g. more generic? Maybe yourname or something like that?

File: extensions/git-branch-name-generator/src/index.tsx [hunk 2]
```diff
@@ -1,56 +1,73 @@
...
+        id="prefixDate"
+        title="Prefix Date"
+        label="Add today's date as a prefix (e.g. 2024-06-05-)"
+        storeValue
+      />
+      <Form.Checkbox
+        id="includeGitCommand"
+        title="Copy Git Command"
+        label="Include `git checkout -b` when copying to clipboard"
+        storeValue
```
Comment (michaelbonner) [inline extensions/git-branch-name-generator/src/index.tsx:69]: The checkboxes make sense to use `storeValue`, but the `description` and `prefix` fields should reset after submitting in my opinoin.

File: extensions/git-branch-name-generator/src/index.tsx [hunk 3]
```diff
@@ -1,56 +1,73 @@
...
+      />
+      <Form.Checkbox
+        id="prefixDate"
+        title="Prefix Date"
+        label="Add today's date as a prefix (e.g. 2024-06-05-)"
+        storeValue
+      />
+      <Form.Checkbox
+        id="includeGitCommand"
```
Comment (michaelbonner) [inline extensions/git-branch-name-generator/src/index.tsx:66]: This currently doesn't do anything in this command. This needs to be wired up in the onSubmit

**Review by michaelbonner — CHANGES_REQUESTED** [review]
There's a handful of things that I think should be changed. Let me know if any of them don't make sense

File: extensions/git-branch-name-generator/src/index.tsx [hunk 4]
```diff
@@ -1,56 +1,73 @@
...
-            .replace(/^-+/, "")
-            // trim dashes from the end
-            .replace(/-+$/, "")
-            // only take the first 100 characters
-            .substring(0, 100)
-        : "";
-      const fullBranchName = `${branchType}${branchNameAsSlug}`;
-      Clipboard.copy(fullBranchName);
-      await showHUD(`Copied to clipboard: ${fullBranchName}`);
-      popToRoot({ clearSearchBar: true });
```
Comment (michaelbonner) [inline extensions/git-branch-name-generator/src/index.tsx:30]: Can we add `popToRoot({ clearSearchBar: true })` back? It's pretty annoying that it persists after I've generated a branch name

File: extensions/git-branch-name-generator/src/index.tsx [hunk 5]
```diff
@@ -1,56 +1,73 @@
...
+      <Form.Dropdown id="branchType" title="Branch Type" storeValue>
+        <Form.Dropdown.Item value="" title="-" />
+        <Form.Dropdown.Item value="fix/" title="fix (bugs or issues)" />
+        <Form.Dropdown.Item value="feat/" title="feat (add new features)" />
+        <Form.Dropdown.Item value="refactor/" title="refactor (improve efficiency)" />
+        <Form.Dropdown.Item value="chore/" title="chore (non-production changes)" />
+        <Form.Dropdown.Item value="docs/" title="docs (add or modify documentation)" />
       </Form.Dropdown>
-      <Form.TextField title="Branch name" placeholder="Change the photo" {...itemProps.branch_name} />
+      <Form.TextField id="description" title="Description" placeholder="Description of this branch" storeValue />
```
[the remark this answers is not in the document]
Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/index.tsx:46]: The intention here is:

When a user wants to start over, they can delete it.
But if the user accidentally closes the tab while writing, they have to start over, which could be a long sentence.

File: extensions/git-branch-name-generator/src/index.tsx [hunk 6]
```diff
@@ -1,56 +1,73 @@
...
+          <Action.SubmitForm title="Branch Name to Clipboard" onSubmit={handleSubmit} />
         </ActionPanel>
       }
     >
-      <Form.Dropdown title="Type of Branch" {...itemProps.type_of_branch}>
-        <Form.Dropdown.Item value="" title="" />
-        <Form.Dropdown.Item value="bug" title="Bug" />
-        <Form.Dropdown.Item value="chore" title="Chore" />
-        <Form.Dropdown.Item value="feature" title="Feature" />
+      <Form.Dropdown id="branchType" title="Branch Type" storeValue>
```
[the remark this answers is not in the document]
Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/index.tsx:38]: Got it, do you know how to let the user add custom dropdowns?
IMO letting user choose the convention could also be a win.

File: extensions/git-branch-name-generator/src/random.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm title="Branch Name to Clipboard" onSubmit={handleSubmit} />
+        </ActionPanel>
+      }
+    >
+      <Form.TextField id="prefix" title="Prefix" placeholder="Add prefix of the branch: (e.g. jwn/~)" storeValue />
```
Comment (devjiwonchoi) [inline extensions/git-branch-name-generator/src/random.tsx:29]:
```suggestion
      <Form.TextField id="prefix" title="Prefix" placeholder="Add prefix of the branch: (e.g. prefix/~)" storeValue />
```

Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/random.tsx:29]: How about this?

File: extensions/git-branch-name-generator/src/index.tsx (hunk 2 above)
Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/index.tsx:69]: For description:

> The intention here is:
>
> When a user wants to start over, they can delete it.
> But if the user accidentally closes the tab while writing, they have to start over, which could be a long sentence.

For prefix:

IMO the date prefix is like a personal convention, so I'd keep it.

File: extensions/git-branch-name-generator/src/index.tsx (hunk 3 above)
Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/index.tsx:66]: Could you clarify

> This currently doesn't do anything in this command

File: extensions/git-branch-name-generator/src/index.tsx (hunk 1 above)
Comment (devjiwonchoi) [inline extensions/git-branch-name-generator/src/index.tsx:50]:
```suggestion
        placeholder="Add a custom prefix for the branch: (e.g. prefix/)"
```

Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/index.tsx:50]: How about this?

File: extensions/git-branch-name-generator/src/index.tsx [hunk 8]
```diff
@@ -1,56 +1,73 @@
...
+  snakeCase: boolean;
+  prefixDate: boolean;
 };
 
+function slugify(content: string, separator: string) {
+  return content.toLowerCase().replaceAll(" ", separator);
+}
+
 export default function Command() {
-  const { handleSubmit, itemProps } = useForm<Values>({
```
[the remark this answers is not in the document]
Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/index.tsx:10]: reverted it thanks!

File: extensions/git-branch-name-generator/src/index.tsx (hunk 4 above)
Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/index.tsx:30]: yea got it back :)

File: extensions/git-branch-name-generator/src/random.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/random.tsx:29]:
```suggestion
      <Form.TextField id="prefix" title="Prefix" placeholder="Add prefix of the branch: (e.g. prefix/~)" storeValue />
```

[the remark this answers is not in the document]
Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/random.tsx:29]: How about this?

File: extensions/git-branch-name-generator/src/index.tsx (hunk 1 above)
Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/index.tsx:76]:
```suggestion
        placeholder="Add a custom prefix for the branch: (e.g. prefix/)"
```

Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/index.tsx:76]: How about this?

File: extensions/git-branch-name-generator/LICENSE [hunk 9]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2024 Michael Bonner, Shu Ding, and Jiwon Choi
```
Comment (devjiwonchoi) [inline extensions/git-branch-name-generator/LICENSE:3]: Added Shu Ding credit for the [random branch generator](https://github.com/shuding/random-branch) (got permission)

File: extensions/git-branch-name-generator/src/index.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (devjiwonchoi) [reply extensions/git-branch-name-generator/src/index.tsx]: Added one, currently we can clear out the whole custom types but not by each. If you have better idea please lmk!

**Comment by devjiwonchoi** [comment]
The failure of CI `Extensions / Check (pull_request)` is weird. It says that `showHUD` can take 1 arguments.

```
ℹ️ info  -  entry points [src/index.tsx src/random.tsx]
ℹ️ info  -  compiled entry points
ℹ️ info  -  generated extension's TypeScript definitions
src/index.tsx(57,64): error TS2554: Expected 1 arguments, but got 2.
src/random.tsx(19,64): error TS2554: Expected 1 arguments, but got 2.
TypeScript check failed (exit status 2)
Error: TypeScript check failed (exit status 2)
Error: Process completed with exit code 1.
```

@mil3na Is there a way I can reproduce that type-check?

**Comment by mil3na** [comment]
Have you run eslint or prettier?

**Comment by devjiwonchoi** [comment]
@mil3na If I reopen https://github.com/raycast/extensions/pull/12780 and leave only the "random" part, with no UI just straight to copying to clipboard which differs from @michaelbonner's extension, could we separate the two?

**Comment by mil3na** [comment]
What do you mean by "separate the two"? Maybe open a new clean PR would be best anyway.

**Pull request closed without merging**